html/admin/api/projects/assets/setStatusByTag.php

```diff
@@ -0,0 +1,23 @@
+<?php
+require_once __DIR__ . '/../../apiHeadSecure.php';
+
+if (!$AUTH->instancePermissionCheck("PROJECTS:PROJECT_ASSETS:EDIT:ASSIGNMENT_STATUS") or !isset($_POST['projects_id']) or !isset($_POST['assetsAssignments_status']) or !isset($_POST['text']) or strlen($_POST['text']) < 1) finish(false, ["message" => "Missing required fields","code"=>"MISSINGFIELDS"]);
+
+$DBLIB->where("assets_deleted",0);
+$DBLIB->where("assets_tag", $_POST["text"]);
+$asset = $DBLIB->getone("assets",["assets.assets_id"]);
+if ($asset and $asset['assets_id'] != null) {
+    $DBLIB->where("assets_id", $asset['assets_id']);
+    $DBLIB->where("projects.projects_id", $_POST['projects_id']);
+    $DBLIB->where("projects.instances_id", $AUTH->data['instance']['instances_id']);
+    $DBLIB->where("projects.projects_deleted", 0);
+    $DBLIB->join("projects", "assetsAssignments.projects_id=projects.projects_id", "LEFT");
+    $DBLIB->update("assetsAssignments", ["assetsAssignmentsStatus_id" => $_POST['assetsAssignments_status']], 1);
```

Comment:
Probs need to check that the assetsAssignments_status exists before setting it, although the rest of the status pages will fail-safe with an unknown value

Reply:
Can do, though looks like the normal `projects/assets/setStatus.php` doesn't have a check there either (is there a reason there isn't a foreign key constraint on `assetsAssignmentsStatus_id`?)

Reply:
Legacy reasons :)

Reply:
I agree with both of you - I think good to check the status id exists in that instance first (because the foreign key can't do that), but then there should be a foreign key too (which I've added, hope you don't mind me pushing to your branch but as its just one new file hopefully will be okay)